jbmc/src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -642,20 +642,39 @@ static irep_idt get_if_cmp_operator(const irep_idt &stmt)
   throw "unhandled java comparison instruction";
 }
 
-static member_exprt to_member(const exprt &pointer, const exprt &fieldref)
+static member_exprt to_member(
+  const exprt &pointer, const exprt &fieldref, const namespacet &ns)
 {
   struct_tag_typet class_type(fieldref.get(ID_class));
 
-  const typecast_exprt pointer2(pointer, java_reference_type(class_type));
+  const exprt typed_pointer = typecast_exprt::conditional_cast(
+    pointer, java_reference_type(class_type));
 
-  dereference_exprt obj_deref=checked_dereference(pointer2, class_type);
+  const irep_idt &component_name = fieldref.get(ID_component_name);
```

Comment:
Is fieldref assumed to be a `member_exprt`? It would be better to make that explicit.

Reply:
It isn't `member_exprt`, but rather the nameless expression that describes a `getfield` or `setfield` operand produced by the Java bytecode parser